brainsets/utils/openneuro/dataset.py

```
@@ -0,0 +1,360 @@
+        if num <= 9999:
+            return f"ds{num:06d}"
+        raise ValueError(
+            f"Dataset ID '{dataset_id}' has too many digits. Maximum is 4 digits."
+        )
+
+    raise ValueError(
+        f"Invalid dataset ID format: '{dataset_id}'. Expected numeric ID or 'ds' + digits."
+    )
+
+
+def fetch_all_filenames(dataset_id: str, tag: Optional[str] = None) -> list[str]:
```

Ruff (0.15.1)

[warning] Unused function argument: `tag`

(ARG001)